From 252e2dfd1748e2165d6b538fd0021412c35cf6b8 Mon Sep 17 00:00:00 2001
From: Julian Greilich <j.greilich@gmx.de>
Date: Tue, 19 May 2026 15:19:18 +0200
Subject: a11y iOS: Implement accessibilityExpandedStatus

This leads to the current expanded/collapsed state being announced.

Task-number: QTBUG-146552
Change-Id: I10f19491498e1d5469b235ad1e2233967fffea5f
---
 .../platforms/ios/quiaccessibilityelement.mm     | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git x/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm y/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
index a4b53bff4b0e9d6a905c43d4e8dfdc798653cfbb..468dce44ba279a72e20abdcc849825957af31bbd 100644
--- x/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
+++ y/qtbase/src/plugins/platforms/ios/quiaccessibilityelement.mm
@@ -215,6 +215,22 @@ QT_NAMESPACE_ALIAS_OBJC_CLASS(QMacAccessibilityElement);
     return traits;
 }
 
+- (UIAccessibilityExpandedStatus)accessibilityExpandedStatus
+{
+    QAccessibleInterface *iface = QAccessible::accessibleInterface(self.axid);
+    if (!iface || !iface->isValid()) {
+        qWarning() << "invalid accessible interface for: " << self.axid;
+        return UIAccessibilityExpandedStatusUnsupported;
+    }
+
+    QAccessible::State state = iface->state();
+    if (!state.expandable)
+        return UIAccessibilityExpandedStatusUnsupported;
+
+    return state.expanded ? UIAccessibilityExpandedStatusExpanded
+                          : UIAccessibilityExpandedStatusCollapsed;
+}
+
 - (BOOL)accessibilityActivate
 {
     QAccessibleInterface *iface = QAccessible::accessibleInterface(self.axid);
